target_postgres/sql_base.py

```diff
@@ -402,6 +394,76 @@ def _canonicalize_identifier(self, name, schema, existing_columns_raw_names, exi
 
         return canonicalized_column_name, canonicalized_typed_column_name
 
+    def add_table(self, connection, schema, metadata):
+        """
+        Create the remote table schema.
+
+        :param connection: remote connection, type left to be determined by implementing class
+        :param schema: TABLE_SCHEMA(local) definition for table to be created
+        :param metadata: additional metadata needed by implementing class
+        :return: None
+        """
+        raise NotImplementedError('`add_table` not implemented.')
+
+    def add_key_properties(self, connection, table_name, key_properties):
+        """
+
+        :param connection: remote connection, type left to be determined by implementing class
+        :param table_name: string
+        :param key_properties: [string, ...]
+        :return: None
+        """
+        raise NotImplementedError('`add_key_properties` not implemented.')
+
+    def add_table_mapping_helper(self, from_path, table_mappings):
+        """
+
+        :param from_path:
+        :param table_mappings:
+        :return: (boolean, string)
+        """
+        from_to = dict([(tuple(mapping['from']), mapping['to']) for mapping in table_mappings])
+
+        ## MAPPING EXISTS
+        if from_path in from_to:
+            return {'exists': True, 'to': from_to[from_path]}
+
+        to_from = dict([(v, k) for k, v in from_to.items()])
+
+        name = SEPARATOR.join(from_path)
+
+        raw_canonicalized_name = self.canonicalize_identifier(name)
+        canonicalized_name = raw_canonicalized_name[:self.IDENTIFIER_FIELD_LENGTH]
+
+        i = 0
+        ## NAME COLLISION
+        while canonicalized_name in to_from:
+            i += 1
+            suffix = SEPARATOR + str(i)
+            canonicalized_name = raw_canonicalized_name[
+                                 :self.IDENTIFIER_FIELD_LENGTH - len(suffix)] + suffix
+
+            # TODO: logger warn
+            ##raise Exception(
```

Comment:
Should this be commented out?

Reply:
Yup. So I think a future refactor (which I have held off on just because it's a #nicetohave) is to move the logger down into `SQLInterface`. Once we do that, we can put back in these logger warns etc.

For this specifically, I think it'd be worthwhile to `warn` the user that a collision has been detected and that we're mucking about with things. The exception that _used_ to be thrown has all of the worthwhile information. It makes more sense in the commit history view of things, but yeah, the TODO is still useful, though I think it'd be more worthwhile as a separate issue etc.